app/bot.py

```diff
@@ -57,10 +59,38 @@ def save_settings(jsonString):
 !status - Sends a self deleting status message with all available server info
 !update - Updates the pinned message from !setup
 !mods - Sends a self deleting message with the currently used mod-list as html file
-!ping - Alive check'''
-
-bot = commands.Bot(command_prefix='!', description=desc)
-
+!ping - Alive check
+!restart - CAUTION: Restarts the server!'''
+
+intents = discord.Intents.default()
+intents.message_content = True
+
+bot = commands.Bot(command_prefix='!', description=desc, intents=intents)
+
+@bot.event
+async def on_ready():
+    print('Logged on')
+
+@bot.command(name="restart")
+async def _restart(ctx):
+    embed = embed_constructor(
+        title='<:warning:1024796220222345216>WARNING<:warning:1024796220222345216>', 
+        text="This command will restart the running arma server without any further questions.\n\
+              If you're sure about that, react with <:elmo:1005177400105107507> within the next 5 seconds.")
+    msg = await ctx.send(embed=embed)
+    await msg.add_reaction("<:elmo:1005177400105107507>")
+    def check(reaction, user):
+        print(reaction)
+        print(user)
+        return user == ctx.author and str(reaction.emoji) == '<:elmo:1005177400105107507>'
+    try:
+        reaction, user = await bot.wait_for('reaction_add', timeout=5.0, check=check)
+    except TimeoutError:
+        await msg.clear_reactions()
+        await ctx.send(f'Restart aborted')
+    else:
+        subprocess.call(["/usr/bin/pkill", "arma3"])
```

Comment:
Could we pass it a SIGHUP to make sure it does a graceful shutdown? Otherwise this leaves us open to save file corruption if the process is killed during IO on autosave.

Kinda like what we have here? https://github.com/mylesagray/Arma3Server/blob/26ac88f1b67132202b0980f1979ce7dca6ccb574/app/launch.py#L155-L159

Reply:
Sure. Since we're just sending the `SIGINT` in our launch.py, i replicated that behaviour in bot.py in [f124adf](https://github.com/mylesagray/Arma3Server/pull/8/commits/f124adf95860e210cc0995bd619a535769054b46)